### PR TITLE
Add Annotations to Action Runs

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This PR adds annotations to Actions runs to clearly highlight important issues from that run.

## 💭 Motivation and context ##

These warnings indicate issues that prevent this project from updating a repository (or running at all) and it would help if they were displayed prominently to help them get noticed.

## 🧪 Testing ##

Automated tests pass. You can see a [sample run](https://github.com/cisagov/action-lineage/actions/runs/1359832089) that has a number of the added annotations.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

### CODE REVIEW

File: src/lineage/entrypoint.py [hunk 1]
```diff
@@ -292,22 +296,32 @@ def main() -> None:
...
         )
+        core.error(
+            "GitHub workspace environment variable must be set. (GITHUB_WORKSPACE)",
+            title="Initialization error",
+        )
         sys.exit(-1)
 
     if repo_query is None:
         logging.fatal(
             "Reository query environment variable must be set. (INPUT_REPO_QUERY)"
```
Comment (jsf9k) [inline src/lineage/entrypoint.py]:
```suggestion
            "Repository query environment variable must be set. (INPUT_REPO_QUERY)"
```

File: src/lineage/entrypoint.py [hunk 2]
```diff
@@ -203,8 +204,11 @@ def push(
     if not repo.permissions.push:
         logging.warning(
             "⚠️ WARNING! Missing 'push' permission on '%s/%s'",
```
Comment (jsf9k) [inline src/lineage/entrypoint.py]: You have two `%s` in the format string but only one argument.
```suggestion
            "⚠️ WARNING! Missing 'push' permission on '%s'",
```

**Review by jsf9k — APPROVED** [review]
I love it!  I did notice two typos.

**Review by dav3r — APPROVED** [review]
Noice! 🥇

**Pull request merged**